App/Syncthing/syncthing_sync_check.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/ash
+
+. /mnt/SDCARD/App/Syncthing/syncthingFunctions.sh
+. /mnt/SDCARD/spruce/scripts/helperFunctions.sh
+
+SYNCTHING_CONFIG_DIR="${SYNCTHING_CONFIG_DIR:-/mnt/SDCARD/App/Syncthing/config}"
+API_ENDPOINT="http://localhost:8384/rest"
+CONFIG_XML="$SYNCTHING_CONFIG_DIR/config.xml"
+CHECK_INTERVAL=2  # Interval between checks in seconds
+MAX_API_RETRIES=15
+API_RETRY_INTERVAL=2
+API_KEY=""
+
+# Function to start network interface
+start_network() {
+    log_message "Starting network interface..."
+    ifconfig lo up
+}
+
+wait_for_syncthing_api() {
+    log_message "Waiting for Syncthing API to become available..."
+    for i in $(seq 1 $MAX_API_RETRIES); do
+        if curl -s -o /dev/null -w "%{http_code}" -H "X-API-Key: $API_KEY" "$API_ENDPOINT/system/status" | grep -q "200"; then
+            log_message "Syncthing API is now available"
+            return 0
+        fi
+        sleep $API_RETRY_INTERVAL
+    done
+    log_message "Syncthing API did not become available within the expected time"
+    return 1
+}
+
+force_rescan() {
+    log_message "Forcing rescan of all folders for upload..."
+    local folders=$(get_folders)
+
+    echo "$folders" | while IFS='|' read -r folder_id folder_label; do
+        curl -s -X POST -H "X-API-Key: $API_KEY" "$API_ENDPOINT/db/scan?folder=$folder_id"
+        log_message "Initiated rescan for folder: $folder_label"
+    done
+}
+
+force_rediscovery() {
+    log_message "Forcing rediscovery of devices..."
+    local response=$(curl -s -X GET -H "X-API-Key: $API_KEY" "$API_ENDPOINT/system/discovery")
+    if [ $? -eq 0 ]; then
+        log_message "Initiated device rediscovery"
+    else
+        log_message "Error: Failed to initiate device rediscovery"
+    fi
+}
+
+stop_network() {
+    log_message "Stopping network interface..."
+    ifconfig lo down
+}
+
+get_folders() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/config/folders" | jq -r '.[] | "\(.id)|\(.label)"'
```

Comment:
Consider a function or even just a string that encapsulates `curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT"` since it gets repeated a lot of times.

Reply:
I'm okay with leaving this as-is, i feel it's more readable this way even though it's a bit repetitive